util/rate_limiter_test.cc

```diff
@@ -416,6 +416,37 @@ TEST_F(RateLimiterTest, LimitChangeTest) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
 }
 
+TEST_F(RateLimiterTest, AvailableByteSizeExhaustTest) {
+  SpecialEnv special_env(Env::Default(), /*time_elapse_only_sleep*/ true);
+  const std::chrono::seconds kTimePerRefill(1);
+
+  std::shared_ptr<RateLimiter> limiter = std::make_shared<GenericRateLimiter>(
+      500 /* bytes */, std::chrono::microseconds(kTimePerRefill).count(),
```

Comment:
nit: might want to make 500 a meaningful constant and refer to it for better semantics in `ASSERT_EQ(500, limiter->GetTotalBytesThrough(Env::IO_USER));`